crates/bevy_core_pipeline/src/motion_blur/mod.rs

```diff
@@ -0,0 +1,112 @@
+//! Per-object motion blur.
+//!
+//! Add the [`MotionBlurBundle`] to a camera to enable motion blur.
+
+use crate::core_3d;
+use bevy_app::{App, Plugin};
+use bevy_asset::{load_internal_asset, Handle};
+use bevy_ecs::{bundle::Bundle, component::Component, schedule::IntoSystemConfigs};
+use bevy_render::{
+    extract_component::{ExtractComponent, ExtractComponentPlugin, UniformComponentPlugin},
+    render_graph::{RenderGraphApp, ViewNodeRunner},
+    render_resource::{Shader, ShaderType, SpecializedRenderPipelines},
+    renderer::RenderDevice,
+    Render, RenderApp, RenderSet,
+};
+
+pub mod node;
+pub mod pipeline;
+
+#[derive(Bundle, Default)]
+pub struct MotionBlurBundle {
+    pub motion_blur: MotionBlur,
+    pub depth_prepass: crate::prepass::DepthPrepass,
+    pub motion_vector_prepass: crate::prepass::MotionVectorPrepass,
+}
+
+/// A component that enables and configures motion blur when added to a camera.
+#[derive(Component, Clone, Copy, Debug, ExtractComponent, ShaderType)]
+pub struct MotionBlur {
+    /// Camera shutter angle from 0 to 1 (0-100%), which determines the strength of the blur.
+    ///
+    /// The shutter angle describes the fraction of a frame that a camera's shutter is open and
+    /// exposing the film/sensor. For 24fps cinematic film, a shutter angle of 0.5 (180 degrees) is
+    /// common. This means that the shutter was open for half of the frame, or 1/48th of a second.
+    /// The lower the shutter angle, the less exposure time and thus less blur. A value greater than
+    /// one is unrealistic and results in an object's blur stretching further than it traveled in
+    /// that frame.
+    pub shutter_angle: f32,
+    /// The upper limit for how many samples will be taken per-pixel. The number of samples taken
+    /// depends on the speed of an object.
+    pub max_samples: u32,
```

Comment:
I noticed the shader returns base color if <2 max samples, so perhaps mention an ok range for users?